src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      for (let i = 0; i < this.length; i++) {
+        for (let k = 0; k < this.length; k++) {
+          if ('' + this[k] > '' + this[k + 1]) {
+            const temp = this[k];
+
+            this[k] = this[k + 1];
+            this[k + 1] = temp;
+          }
+        }
```

Comment:
The inner loop should iterate only until 'this.length - i - 1' to avoid unnecessary comparisons and to avoid checking an undefined value when 'k + 1' equals 'this.length'.